identity-android/src/main/java/com/android/identity/android/direct_access/DirectAccessCredential.kt

```diff
@@ -0,0 +1,166 @@
+            encryptedPresentationData = it.second
+        }
+
+        // Only the leaf constructor should add the credential to the document.
+        if (this::class == DirectAccessCredential::class) {
+            addToDocument()
+        }
+    }
+
+    /**
+     * Constructs a Credential from serialized data.
+     *
+     * @param document the [Document] that the credential belongs to.
+     * @param dataItem the serialized data.
+     */
+    constructor(
+        document: Document,
+        dataItem: DataItem,
+    ) : super(document, dataItem) {
+        docType = dataItem["docType"].asTstr
+        documentSlot = dataItem["documentSlot"].asNumber.toInt()
+        encryptedPresentationData = dataItem["encryptedPresentationData"].asBstr
+        signingCert = X509CertChain.fromDataItem(dataItem["signingCert"])
+    }
+
+    /**
+     * The docType of the credential as defined in
+     * [ISO/IEC 18013-5:2021](https://www.iso.org/standard/69084.html).
+     */
+    val docType: String
+
+    /**
+     * The attestation for the key associated with this credential.
+     *
+     * The application should send this attestation to the issuer which should create
+     * issuer-provided data (if using ISO/IEC 18013-5:2021 this would include the MSO).
+     * Once received, the application should call [Credential.certify] to certify
+     * the [Credential].
+     */
+    val attestation: KeyAttestation
+        get() {
+            return KeyAttestation(signingCert.certificates.first().ecPublicKey, signingCert)
+        }
+
+    val documentSlot: Int
+    private var encryptedPresentationData: ByteArray
+    private val signingCert: X509CertChain
+
+    override fun addSerializedData(builder: MapBuilder<CborBuilder>) {
+        super.addSerializedData(builder)
+        builder.put("docType", docType)
+        builder.put("documentSlot", documentSlot)
+        builder.put("encryptedPresentationData", encryptedPresentationData)
+        builder.put("signingCert", signingCert.toDataItem())
+    }
+
+    // Provisions credential data for a specific signing key request.
+    //
+    // The |credentialData| parameter must be CBOR conforming to the following CDDL:
+    //
+    //   CredentialData = {
+    //     "docType": tstr, // todo remove once applet is updated
+    //     "issuerNameSpaces": IssuerNameSpaces,
+    //     "issuerAuth" : IssuerAuth,
+    //     "readerAccess" : ReaderAccess // todo update applet for name change to "authorizedReaderRoots"
+    //   }
+    //
+    //   IssuerNameSpaces = {
+    //     NameSpace => [ + IssuerSignedItemBytes ]
+    //   }
+    //
+    //   ReaderAccess = [ * COSE_Key ]
+    //
+    // This data will be stored on the Secure Area and used for MDOC presentations
+    // using NFC data transfer in low-power mode.
+    //
+    // The `readerAccess` field contains a list of keys used for implementing
+    // reader authentication. If this list is non-empty, reader authentication
```

Comment:
This sounds backwards. Should this be: "If this list is empty, reader authentication is not required." ?